### PR TITLE
sns_topic_info new module

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

`sns_topic_info` -  new module allowing to get all AWS SNS topics or properties of a specific AWS SNS topic.

Fixes https://github.com/ansible-collections/community.aws/issues/601

Requires https://github.com/ansible-collections/community.aws/pull/879

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

sns_topic_info

### CODE REVIEW

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

File: plugins/modules/sns_topic_info.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        sample: "arn:aws:sns:us-east-2:111111111111:my_topic_name"
+    sns_topic:
+        description: Dict of sns topic details
+        type: complex
+        returned: always
+        contains:
+            delivery_policy:
+                description: Delivery policy for the SNS topic
+                returned: when topic is owned by this AWS account
+                type: str
```
Comment (jatorcasso) [inline plugins/modules/sns_topic_info.py:65]:
```suggestion
                type: dict
```

File: plugins/modules/sns_topic_info.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+                sample: My topic name
+            owner:
+                description: AWS account that owns the topic
+                returned: when topic is owned by this AWS account
+                type: str
+                sample: '111111111111'
+            policy:
+                description: Policy for the SNS topic
+                returned: when topic is owned by this AWS account
+                type: str
```
Comment (jatorcasso) [inline plugins/modules/sns_topic_info.py:83]:
```suggestion
                type: dict
```

File: plugins/modules/sns_topic_info.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            subscriptions_pending:
+                description: Count of pending subscriptions
+                returned: when topic is owned by this AWS account
+                type: str
+                sample: '0'
+            topic_arn:
+                description: ARN of the SNS topic (equivalent to sns_arn)
+                returned: when topic is owned by this AWS account
+                type: str
+                sample: arn:aws:sns:us-east-2:111111111111:ansible-test-dummy-topic
```
Comment (jatorcasso) [inline plugins/modules/sns_topic_info.py:112]: Info module also returns `subscriptions_added`, `subscriptions_existing`, `subscriptions_new`, `subscriptions_purge`, and `topic_type`. Do we want to include these as well?

File: tests/integration/targets/sns_topic/tasks/main.yml [hunk 4]
```diff
@@ -21,6 +20,25 @@
     pause:
       seconds: 10
     when: iam_role is changed
+
```
Comment (jatorcasso) [inline tests/integration/targets/sns_topic/tasks/main.yml:23]:
```suggestion

```

File: tests/integration/targets/sns_topic/tasks/main.yml [hunk 5]
```diff
@@ -21,6 +20,25 @@
     pause:
       seconds: 10
     when: iam_role is changed
+  
+  - name: list all the topics (check_mode)
+    sns_topic_info:
+    check_mode: true
+    register: sns_topic_list
+
```
Comment (jatorcasso) [inline tests/integration/targets/sns_topic/tasks/main.yml:28]:
```suggestion

```

File: tests/integration/targets/sns_topic/tasks/main.yml [hunk 6]
```diff
@@ -21,6 +20,25 @@
...
+  - name: list all the topics (check_mode)
+    sns_topic_info:
+    check_mode: true
+    register: sns_topic_list
+  
+  - name: assert success
+    assert:
+      that:
+          - sns_topic_list is successful
+
```
Comment (jatorcasso) [inline tests/integration/targets/sns_topic/tasks/main.yml:33]:
```suggestion

```

File: tests/integration/targets/sns_topic/tasks/main.yml [hunk 7]
```diff
@@ -21,6 +20,25 @@
...
+  
+  - name: assert success
+    assert:
+      that:
+          - sns_topic_list is successful
+  
+  - name: list all the topics
+    sns_topic_info:
+    register: sns_topic_list
+
```
Comment (jatorcasso) [inline tests/integration/targets/sns_topic/tasks/main.yml:37]:
```suggestion

```

File: tests/integration/targets/sns_topic/tasks/main.yml [hunk 8]
```diff
@@ -36,6 +54,41 @@
   - name: set sns_arn fact
     set_fact:
       sns_arn: '{{ sns_topic_create.sns_arn }}'
+
```
Comment (jatorcasso) [inline tests/integration/targets/sns_topic/tasks/main.yml:57]:
```suggestion

```

File: tests/integration/targets/sns_topic/tasks/main.yml [hunk 9]
```diff
@@ -36,6 +54,41 @@
   - name: set sns_arn fact
     set_fact:
       sns_arn: '{{ sns_topic_create.sns_arn }}'
+  
+  - name: get info on specific topic (check_mode)
+    sns_topic_info:
+      topic_arn: "{{ sns_arn }}"
+    check_mode: true
+    register: sns_topic_info
+
```
Comment (jatorcasso) [inline tests/integration/targets/sns_topic/tasks/main.yml:63]:
```suggestion

```

File: tests/integration/targets/sns_topic/tasks/main.yml [hunk 10]
```diff
@@ -36,6 +54,41 @@
...
+      that:
+      - sns_topic_info is successful
+      - "'result' in sns_topic_info"
+      - sns_topic_info.result["sns_arn"] == "{{ sns_arn }}"
+      - "'sns_topic' in sns_topic_info.result"
+      - "'display_name' in sns_topic_info.result['sns_topic']"
+      - sns_topic_info.result["sns_topic"]["display_name"] == "My topic name"
+      - "'owner' in sns_topic_info.result['sns_topic']"
+      - "'policy' in sns_topic_info.result['sns_topic']"
+
```
Comment (jatorcasso) [inline tests/integration/targets/sns_topic/tasks/main.yml:75]:
```suggestion

```

File: tests/integration/targets/sns_topic/tasks/main.yml [hunk 11]
```diff
@@ -36,6 +54,41 @@
...
+      - "'display_name' in sns_topic_info.result['sns_topic']"
+      - sns_topic_info.result["sns_topic"]["display_name"] == "My topic name"
+      - "'owner' in sns_topic_info.result['sns_topic']"
+      - "'policy' in sns_topic_info.result['sns_topic']"
+  
+  - name: get info on specific topic
+    sns_topic_info:
+      topic_arn: "{{ sns_arn }}"
+    register: sns_topic_info
+
```
Comment (jatorcasso) [inline tests/integration/targets/sns_topic/tasks/main.yml:80]:
```suggestion

```

**Review by jatorcasso — APPROVED** [review]
Added some minor comments about whitespace and documentation - LGTM!

File: plugins/modules/sns_topic_info.py (hunk 1 above)
Comment (alinabuzachis) [reply plugins/modules/sns_topic_info.py:65]: It should be string (to be consistent with sns_topic), I corrected the sample. Thank you for catching this. Same for `delivery_policy`.

File: plugins/modules/sns_topic_info.py (hunk 3 above)
Comment (alinabuzachis) [reply plugins/modules/sns_topic_info.py:130]: Added.

**Comment by alinabuzachis** [comment]
recheck

**Comment by marknet15** [comment]
recheck

**Review by marknet15 — APPROVED** [review]

**Comment by markuman** [comment]
recheck

**Review by markuman — APPROVED** [review]

**Comment by markuman** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by marknet15** [comment]
recheck

**Pull request merged**